Author: unknown
URL: no URL yet
Add functionality to restart dwm.

Index: dwm/dwm.c
===================================================================
--- dwm/dwm.c.orig
+++ dwm/dwm.c
@@ -262,6 +262,7 @@ static void (*handler[LASTEvent]) (XEven
 };
 static Atom wmatom[WMLast], netatom[NetLast];
 static int running = 1;
+static int restart = 0;
 static Cur *cursor[CurLast];
 static Scm scheme[SchemeLast];
 static Display *dpy;
@@ -1268,6 +1269,8 @@ propertynotify(XEvent *e)
 void
 quit(const Arg *arg)
 {
+	if(arg->i)
+		restart = 1;
 	running = 0;
 }
 
@@ -2134,6 +2137,8 @@ main(int argc, char *argv[])
 	setup();
 	scan();
 	run();
+	if(restart)
+		execvp(argv[0], argv);
 	cleanup();
 	XCloseDisplay(dpy);
 	return EXIT_SUCCESS;
