Author: Unknown (extended by Jan Christoph Ebersbach <jceb@e-jc.de)
URL: no URL yet
Shift tags, e.g. cycle through tags.

Index: dwm/shifttags.c
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ dwm/shifttags.c	2014-02-09 15:24:25.076117324 +0100
@@ -0,0 +1,28 @@
+static void
+shifttags(const Arg *arg) {
+	Arg shift;
+	/* wrap around tags when shifting over the ends of tagmask */
+	/* only activate wrapping when the number of tags fits twice into an
+	 * unsigned int */
+	Bool wrap = LENGTH(tags) <= (sizeof(shift.ui) * 4) ? True : False;
+
+	if(arg->i == 0)
+		return;
+
+	if(arg->i > 0)
+		shift.ui = selmon->tagset[selmon->seltags] << arg->i;
+	else {
+		if(wrap) {
+			shift.ui = selmon->tagset[selmon->seltags] << LENGTH(tags);
+			shift.ui = shift.ui >> (-1 * arg->i);
+		}
+		else
+			shift.ui = selmon->tagset[selmon->seltags] >> (-1 * arg->i);
+	}
+	if(wrap)
+		shift.ui |= shift.ui >> LENGTH(tags);
+
+	if(!(shift.ui & TAGMASK))
+		return;
+	view(&shift);
+}
